logs/parse.go

```diff
@@ -541,13 +511,69 @@ func ParseLogLineWithPrefix(prefix string, line string) (logLine state.LogLine,
 	return
 }
 
+func getOccurredAt(timePart, timeFormat, timeFormatAlt string, tz *time.Location) time.Time {
+	if tz != nil {
+		result, err := time.ParseInLocation(timeFormat, timePart, tz)
+		if err == nil {
+			return result
+		}
+		if timeFormatAlt != "" {
+			// Ensure we have the correct format remembered for ParseInLocation call that may happen later
+			timeFormat = timeFormatAlt
+			result, err := time.ParseInLocation(timeFormat, timePart, tz)
+			if err == nil {
+				return result
+			}
+		}
```

Comment:
This mirroring the logic below is pretty awkward. I don't know if there's a good way to just ignore the time zone in this case.

Reply:
When location is passed with `time.ParseInLocation`, it'll be unclear how the timezone info within the original `timeFormat` part will be treated.

> ParseInLocation is like Parse but differs in two important ways. First, in the absence of time zone information, Parse interprets a time as UTC; ParseInLocation interprets the time as in the given location. Second, when given a zone offset or abbreviation, Parse tries to match it against the Local location; ParseInLocation uses the given location.

In short:
* not timezone info => interprets the time as in the given location
* no zone offset or abbreviation => uses the given location

With this, I tested with several cases to see how it behaves, especially when the offset/abbreviation is different from the given location's one.
Summarizing, it'll try to respect the original `timeFormat`' s offset/abbreviation over the given location, then it's not really going well (esp. abbr case as the offset will be 0, which is the most important part to calculate unix time). Honestly, I don't even know if such case exists, but I figured that I'd raise this as concern.
With this, when we have a location, I'd say that we can actually try to drop offset/abbreviation like you mentioned from `timeFormat` then use the unique formatting like `const shortForm = "2006-01-02 15:04:05"`, then completely rely on a location info to set timezone.
One thing I noticed: we already kinda do this pattern with rsyslog, as it's using `2006 Jan  2 15:04:05` as a format. With this new change, previously above was always assumed as UTC, but potentially it turns into some other timezone? I might just be thinking some impossible, so feel free to let me know "no, Keiko, it won't happen".

-----

### Case 1-1: with offset (but not matching to the location)

```go
const longForm = "2006-01-02 15:04:05 -0700"
tz, _ := time.LoadLocation("Europe/London")
time.ParseInLocation(longForm, "2023-07-01 00:00:00 -0500", tz)
fmt.Println(t)
// returns 2023-07-01 00:00:00 -0500 -0500
// Unix time: 1688187600
// zone: "", offset: -18000 (-5 * 3600)
```

### Case 1-2: with offset (matching to the location)

```go
const longForm = "2006-01-02 15:04:05 -0700"
tz, _ := time.LoadLocation("Asia/Tokyo")
t, _ := time.ParseInLocation(longForm, "2023-07-01 00:00:00 +0900", tz)
fmt.Println(t)
// returns 2023-07-01 00:00:00 +0900 JST
// Unix time: 1688137200
// zone: JST, offset: 32400 (9 * 3600)
```

### Case 2-1: with abbreviation (but not matching to the location)

```go
const longForm = "2006-01-02 15:04:05 MST"
tz, _ := time.LoadLocation("Europe/London")
t, _ := time.ParseInLocation(longForm, "2023-07-01 00:00:00 JST", tz)
fmt.Println(t)
// returns 2023-07-01 00:00:00 +0000 JST
// Unix time: 1688169600
// zone: JST, offset: 0
```

### Case 2-2: with abbreviation (matching to the location)

```go
const longForm = "2006-01-02 15:04:05 MST"
tz, _ := time.LoadLocation("Europe/London")
t, _ := time.ParseInLocation(longForm, "2023-07-01 00:00:00 BST", tz)
fmt.Println(t)
// returns 2023-07-01 00:00:00 +0100 BST
// Unix time: 1688166000
// zone: BST, offset: 3600
```

Reply:
Thanks for the feedback and analysis. I think this suggests we should strip off the timezone before processing the timestamps when we have a location from `log_timezone`. I'll update the code to do that.